### ISSUE TITLE
[Feature Request] RWKV 7 Weight Decay Not Log Option

### ISSUE THREAD

**Issue by lucassunalt** [fla-org/flash-linear-attention#415]
### Feature Request

Please add an option for RWKV 7's weight decay not to be limited to being in log space, by having an additional argument `log_decay=True` in the RWKV 7 chunk_rwkv7 function. 

### Motivation

I wish the decay term could be negative or exactly zero, which would be helpful in some situations. However, the decay term is in log space normally, which forces the decay term to be greater than zero, limiting its numerical range.

### Your Contribution

I am admittedly not used to working with Triton. However, I believe this is a minor change and only requires adding an if statement.

**Comment by zhiyuan1i** [comment]
I have some questions that I don't understand you, you can manually pass in `w = -w_orgi.exp()`, in my understanding, this is the same as your needs.

**Comment by lucassunalt** [comment]
Sorry for any confusion. 

Currently, the `w` term I pass in gets exponentiated inside the kernel, so it is always positive. If I pass in `decay_term = x` then the diagonal matrix will have `diagonal = exp(x)`. 

I propose to optionally disable this exponentiation step so that the diagonal matrix can also have `diagonal = x`, allowing the diagonal to have negative entries.

**Comment by lucassunalt** [comment]
So to take the `fused_recurrent` implementation as an example, on Line 96:

```python
b_h = exp(b_w)[None, :] * b_h + ...
```

I propose to optionally allow the fused_recurrent and chunk_rwkv7 kernels to also run

```python
b_h = b_w[None, :] * b_h + ...
```

if `log_decay=False` is passed into the function.

**Comment by zhiyuan1i** [comment]
Thank you for the clarification; I now fully understand your perspective The changes are not particularly challenging, though I am somewhat hesitant about the performance. Regarding applicable scenarios, could you provide further explanation on instances where significant performance improvements have been verified using naive PyTorch (by manual backward)? Simply delete all exponential functions in the following code: https://github.com/fla-org/flash-linear-attention/blob/main/fla/ops/rwkv7/RWKV7(Goose).md.


I may need to consult with @yzhangcs . What is your opinion?

**Comment by yzhangcs** [comment]
@lucassunalt Do you need this feature for both training and inference?

**Comment by lucassunalt** [comment]
Preferably both. Thanks!

**Comment by zhiyuan1i** [comment]
> Preferably both. Thanks!

I believe decoding is relatively straightforward https://github.com/fla-org/flash-linear-attention/tree/log-decay. However, when it comes to training, we have been using a chunk kernel so far, and it is a significant task. This is because all cumsum and small kernels need to be modified. I must ensure they are mathematically consistent in both forward and backward computations.

Providing some benefits achieved through this method could greatly inspire us to implement those changes effectively.

**Comment by lucassunalt** [comment]
The benefit is that chunk_rwkv7 will be more generalized, as it can represent more state transitions. I don't have any empirical results yet, so its fine if this doesn't get fully implemented right now.
